security/experimental_authenticators.rst

```diff
@@ -0,0 +1,434 @@
+login and an API token authentication). In these cases, it is now required
+to configure the *authentication entry point*. The entry point is used to
+generate a response when the user is not yet authenticated. This can be
+used for instance to redirect the user to the login page.
+
+You can configure this using the ``entry_point`` setting:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            enable_authenticator_manager: true
+
+            # ...
+            firewalls:
+                main:
+                    # allow authentication using a form or HTTP basic
+                    form_login: ~
+                    http_basic: ~
+
+                    # configure the form authentication as the entry point for unauthenticated users
+                    entry_point: form_login
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <config enable-authenticator-manager="true">
+                <!-- ... -->
+
+                <!-- entry-point: configure the form authentication as the entry
+                                  point for unauthenticated users -->
+                <firewall name="main"
+                    entry-point="form_login"
+                >
+                    <!-- allow authentication using a form or HTTP basic -->
+                    <form-login/>
+                    <http-basic/>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        use Symfony\Component\Security\Http\Firewall\AccessListener;
+
+        $container->loadFromExtension('security', [
+            'enable_authenticator_manager' => true,
+
+            // ...
+            'firewalls' => [
+                'main' => [
+                    // allow authentication using a form or HTTP basic
+                    'form_login' => null,
+                    'http_basic' => null,
+
+                    // configure the form authentication as the entry point for unauthenticated users
+                    'entry_point' => 'form_login'
+                ],
+            ],
+        ]);
+
+.. note::
+
+    You can also create your own authentication entry point by creating a
+    class that implements
+    :class:`Symfony\\Component\\Security\\Http\\EntryPoint\\AuthenticationEntryPointInterface`.
+    You can then set ``entry_point`` to the FQCN (e.g.
```

Comment:
Hmm, I'm wondering if FCQN should say "service id" to be more accurate. But we would still show a class name in the example below.